### PR TITLE
Fix drawing tests for the new TAEF world and fix OS X tests.

### LINKED ISSUE #1941: Investigate TAEF interop with OSX tests
Make sure the new world didn't break OSX tests.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
Additionally, this pull request fixes:
* The combined TAEF/CLI entrypoint in drawing tests.
* The file output medium for drawing tests; it'll use WEX's file logger now!

Fixes #1941, #1940.

### CODE REVIEW

File: build/Tests/UnitTests/CoreGraphics.Drawing/CoreGraphics.Drawing.UnitTests.vcxproj [hunk 1]
```diff
@@ -232,7 +232,7 @@
   </ItemDefinitionGroup>
   <ItemGroup>
     
-    <ClangCompile Include="$(StarboardBasePath)\tests\unittests\CoreGraphics.Drawing\EntryPoint.cpp" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\unittests\CoreGraphics.Drawing\TAEFEntryPoint.cpp" />
```
Comment (bbowman) [inline build/Tests/UnitTests/CoreGraphics.Drawing/CoreGraphics.Drawing.UnitTests.vcxproj]: >TAEFEntryPoint [](start = 85, length = 14)

Nice! much cleaner.

File: tests/frameworks/gtest/include/gtest/gtest-param-test.h [hunk 2]
```diff
@@ -1406,6 +1406,8 @@ internal::CartesianProductHolder10<Generator1, Generator2, Generator3,
       GTEST_TEST_CLASS_NAME_(test_case_name, test_name)::AddToRegistry(); \
   void GTEST_TEST_CLASS_NAME_(test_case_name, test_name)::TestBody()
 
+# define TEST_P(test_case_name, test_name) GTEST_TEST_P_(test_case_name, test_name, test_case_name)
```
Comment (bbowman) [inline tests/frameworks/gtest/include/gtest/gtest-param-test.h]: >TEST_P [](start = 9, length = 6)

Just so I understand this, you are modifying the gtest api to have another level of indirection here so we can do the interstitial doodad?

Comment (DHowett-MSFT) [inline tests/frameworks/gtest/include/gtest/gtest-param-test.h]: yep!

Comment (aballway) [inline tests/frameworks/gtest/include/gtest/gtest-param-test.h]: thank you

File: tests/frameworks/include/TestFramework.h [hunk 3]
```diff
@@ -20,8 +20,19 @@
...
 
+#ifndef IF_NOT_EXISTS_BEGIN
+#ifdef _MSC_VER
+#define IF_NOT_EXISTS_BEGIN(name) __if_not_exists(name) {
+#define IF_NOT_EXISTS_END }
+#else
+#define IF_NOT_EXISTS_BEGIN(name)
+#define IF_NOT_EXISTS_END
+#endif
+#endif
```
Comment (bbowman) [inline tests/frameworks/include/TestFramework.h]: solid.

File: tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp [hunk 4]
```diff
@@ -26,6 +26,21 @@
...
+#define LOG_TEST_FILE(...) WEX::Logging::Log::File(__VA_ARGS__)
+std::wstring _CFStringToWString(CFStringRef string) {
+    const UniChar* characters16 = CFStringGetCharactersPtr(string);
+    if (characters16) {
+        return {characters16, characters16 + CFStringGetLength(string)};
+    }
+
+    static std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+    return converter.from_bytes(CFStringGetCStringPtr(string, kCFStringEncodingUTF8));
+}
```
Comment (bbowman) [inline tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp]: add a newline before this guy? I was confused what the log test thing did until I realized this was separate.

File: tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp [hunk 5]
```diff
@@ -26,6 +26,21 @@
 #import "CGContextInternal.h"
 #endif
 
+#ifdef TARGET_OS_MAC
+#define LOG_TEST_FILE(...)
```
Comment (bbowman) [inline tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp]: no record property?

Comment (DHowett-MSFT) [inline tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp]: nah, not necessary to record the property in an xml file we don't generate for a platform we don't want to diff on :)

Comment (bbowman) [inline tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp]: fair. Would be useful to move this macro to TestFramework for the benefit of other tests?

---
In reply to: [100652177](https://github.com/Microsoft/WinObjC/pull/1948#discussion_r100652177) [](ancestors = 100652177)

**Comment by bbowman** [comment]
>                                                                    GetTestFullName().c_str(),

Is this not wrong now? It would pick up the interstitial name as well right?

---
Refers to: tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp:80 in c7d625e. [](commit_id = c7d625ec6ca67ef2a5981facd24adb1a58271094, deletion_comment = False)

**Comment by bbowman** [comment]
>                                                                    GetTestFullName().c_str(),

I"m a dummy. I see how this works. polymorphism is fun.

---
In reply to: [279105423](https://github.com/Microsoft/WinObjC/pull/1948#issuecomment-279105423) [](ancestors = 279105423)

---
Refers to: tests/UnitTests/CoreGraphics.drawing/DrawingTest.cpp:80 in c7d625e. [](commit_id = c7d625ec6ca67ef2a5981facd24adb1a58271094, deletion_comment = False)

**Comment by DHowett-MSFT** [comment]
@bbowman Since we changed the gtest API, the interstitial is only used as the abse class. Its name does not get embedded anywhere 🗡

**Comment by bbowman** [comment]
yep. I see that now. sweet sword.

---
In reply to: [279105585](https://github.com/Microsoft/WinObjC/pull/1948#issuecomment-279105585) [](ancestors = 279105585)

**Comment by bbowman** [comment]
:shipit:

**Pull request merged**